dali/ft/filecopy.cpp

```diff
@@ -816,6 +816,16 @@ void FileSprayer::beforeTransfer()
         checker.For(targets.ordinality(), 25, true, true);
     }
 
+    int umask = -1;
+    if (options->hasProp("@umask"))
+    {
+        StringBuffer umaskStr;
+        options->getProp("@umask", umaskStr);
+        umask = (int)strtol(umaskStr.str(), NULL, 8);
```

Comment:
probably worth validating result, e.g. checking errno, issuing a logged warning and leaving umask as -1